cmd/object-handlers.go

```diff
@@ -1757,8 +1757,23 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 		md5hex = "" // Do not try to verify the content.
 		sha256hex = ""
 	}
-
-	hashReader, err := hash.NewReader(reader, size, md5hex, sha256hex, actualSize)
+	var hashReader *hash.Reader
+	// SSE-KVM and SSE-C if not request Content-Md5. Simple use md5.Sum(uuid) as etag.
+	if !etag.ContentMD5Requested(r.Header) && (crypto.S3KMS.IsRequested(r.Header) || crypto.SSEC.IsRequested(r.Header)) {
+		hashReader, err = hash.NewReaderWithOpts(ctx, reader, hash.Options{
+			Size:       size,
+			MD5Hex:     md5hex,
+			SHA256Hex:  sha256hex,
+			ActualSize: actualSize,
+			DisableMD5: false,
+			ContentMD5: hash.ContentMD5Opt{
+				NotRequest: true,
```

Comment:
Instead of Not request name it as 

```suggestion
				Requested: false,
```

Reply:
@jiuker In general, it is good practice to never name your bools as a negative. 

When you do, you easily end up with things that are hard to read.

`if !NotRequested {}` takes a lot harder to read than `if Requested` - and it gets worse when you combine it and `&&` and `||` in a longer sequence.

There are exceptions. But that is what we should try to do when we can.

Naming is hard - even when you've been programming a long time. And as for most of us where English is a second or third language it is even harder.